### PR TITLE
Clarifying that synchronous Subscribers are RECOMMENDED to request() …

### PR DESCRIPTION
**Pull request by viktorklang**
…at the end of signal processing

See #429

### CODE REVIEW

**Comment by rkuhn** [comment]
To me adding it as proposed is fine.

**Comment by viktorklang** [comment]
@rkuhn Thanks Dr. Kuhn!

**Comment by DougLea** [comment]
looks ok, but it would still be nice to emphasize here that subscribers should request the max they can receive, not necessarily one.

File: README.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ public interface Subscriber<T> {
 | ID                        | Rule                                                                                                   |
 | ------------------------- | ------------------------------------------------------------------------------------------------------ |
 | <a name="2.1">1</a>       | A `Subscriber` MUST signal demand via `Subscription.request(long n)` to receive `onNext` signals. |
-| [:bulb:](#2.1 "2.1 explained") | *The intent of this rule is to establish that it is the responsibility of the Subscriber to signal when, and how many, elements it is able and willing to receive.* |
+| [:bulb:](#2.1 "2.1 explained") | *The intent of this rule is to establish that it is the responsibility of the Subscriber to signal when, and how many, elements it is able and willing to receive. To avoid signal reordering caused by reentrant Subscription methods, it is strongly RECOMMENDED for synchronous Subscriber implementations to invoke Subscription methods at the very end of any signal processing.* |
```
[the remark this answers is not in the document]
Comment (ktoso) [reply README.md]: Not really part of the rule itself I think... having it here sounds good. Wording is good too.

**Comment by viktorklang** [comment]
I think that a recommendation for request(…) should go into the Intent section of 2:1—https://github.com/reactive-streams/reactive-streams-jvm#2.1

Please open a PR with a suggested wording

**Comment by viktorklang** [comment]
(I mean that it is a good addition and it deserves its own PR :))

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (viktorklang) [reply README.md]: Thanks @ktoso!

Comment (davidmoten) [inline README.md]: > when, and how many, elements it is able

The commas need to disappear on this one because you cannot say "when elements it is able".

Comment (viktorklang) [reply README.md]: @davidmoten That's a good improvement, thank you!

File: README.md [hunk 2]
```diff
@@ -131,7 +131,7 @@ public interface Subscriber<T> {
 | ID                        | Rule                                                                                                   |
 | ------------------------- | ------------------------------------------------------------------------------------------------------ |
 | <a name="2.1">1</a>       | A `Subscriber` MUST signal demand via `Subscription.request(long n)` to receive `onNext` signals. |
-| [:bulb:](#2.1 "2.1 explained") | *The intent of this rule is to establish that it is the responsibility of the Subscriber to signal when, and how many, elements it is able and willing to receive.* |
+| [:bulb:](#2.1 "2.1 explained") | *The intent of this rule is to establish that it is the responsibility of the Subscriber to decide when and how many elements it is able and willing to receive. To avoid signal reordering caused by reentrant Subscription methods, it is strongly RECOMMENDED for synchronous Subscriber implementations to invoke Subscription methods at the very end of any signal processing.* |
```
Comment (viktorklang) [inline README.md]: @reactive-streams/contributors Please have a look at this rewording suggested by @davidmoten: https://github.com/reactive-streams/reactive-streams-jvm/pull/433/commits/8360e3453e6bcba3190ed51cd5d56335f8ddfe61

**Comment by ktoso** [comment]
> (I mean that it is a good addition and it deserves its own PR :))

Sorry, was busy over ScalaDays, I can PR the mini improvement today, I suppose "on top of" this PR.

File: README.md (hunk 2 above)
Comment (ktoso) [reply README.md]: Sounds good to me

**Comment by ktoso** [comment]
> I think that a recommendation for request(…) should go into the Intent section of 2:1—https://github.com/reactive-streams/reactive-streams-jvm#2.1
> 
> Please open a PR with a suggested wording

PSed https://github.com/reactive-streams/reactive-streams-jvm/pull/435

**Comment by viktorklang** [comment]
@reactive-streams/contributors Thanks for reviewing!

**Pull request merged**